src/Field/Field.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import styled from 'styled-components'
+import { darken } from 'polished'
+
+import { Icon } from '../Icon'
+import { Text } from '../Text'
+import { Box } from '../Box'
+import { theme } from '../theme'
+
+type Props = {
+  children: React.ReactElement
+  showCaret?: boolean
+  renderAsTitle?: boolean
+  className?: string
+  id: string
+  error?: boolean // nb this needs to be optional for legacy support - but we will want to enforce in  the future
+  label?: string
+  outlined?: boolean
+  value: string
+  trailingIcon?: string
+  errorMsg?: string
+  dropdownKey?: string // nb only used by dropdown component but required on content in that context
+  required?: boolean // also optional for now to maintain backwards compaibility
+  fullHeight?: boolean // used to make text areas full size
+}
+
+export const Field = ({
+  children,
+  renderAsTitle,
+  className = '',
+  error,
+  id,
+  label,
+  outlined = false,
+  value,
+  trailingIcon,
+  errorMsg,
+  dropdownKey,
+  required,
+  showCaret,
+  fullHeight = false,
+}: Props) => {
+  return (
+    <Container className={className}>
+      {label && (
+        <Box mb={outlined ? '4px' : '0px'}>
+          {renderAsTitle ? (
+            <Title>{label}</Title>
+          ) : (
+            <Text tag="label" color="subtext" typo="label" htmlFor={id}>
+              {label}
+            </Text>
+          )}
+          {required && (
+            <Text tag="label" color="error" typo="label">
+              *
+            </Text>
+          )}
+        </Box>
+      )}
+      <Content
+        fullHeight={fullHeight}
+        value={value}
+        outlined={outlined}
+        error={error}
+        key={dropdownKey ?? null}
+      >
+        {children}
+        {showCaret && (
+          <Caret outlined={outlined}>
+            <Icon render="caret" color="subtext" size={24} />
+          </Caret>
+        )}
+      </Content>
+      {trailingIcon && <Icon render={trailingIcon} color="subtext" />}
+
+      {error && <ErrorBox>{errorMsg}</ErrorBox>}
+    </Container>
+  )
+}
+
+const Container = styled(Box)<{ className: string }>`
+  display: flex;
+  flex-direction: column;
+  position: relative;
+`
+
+const Content = styled.div<{
+  error?: boolean
+  outlined: boolean
+  value?: string
+  fullHeight?: boolean
+}>`
```

Comment:
Hmm, not sure I agree - since the idea behind this is to make sure we have uniform styling across all fields, I think it makes sense to enforce things like height and border on the field level. We could abstract into it's own component as you suggested, but I think at this point that fits into the over abstraction category.  It would also break the philosophy behind doing 

```
<Field>
<whateverInput/>
</Field>
```

as We'd now need a separate Content (or whatever it was called) wrapper as well

Reply:
Yeah I don't think they are input related. They are spacing and layout changes based on how the input behave so agree that should live in `Field`

Reply:
I guess if it used only for input, we can keep it there but I feel that `Field` could be used for other types of control like the date picker or a custom components